zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java

```diff
@@ -62,6 +65,38 @@ public CompletionStage<PendingIncidentUpdateBatch> getPendingIncidentsBatch(
         .thenApplyAsync(this::createPendingIncidentBatch, executor);
   }
 
+  @Override
+  public CompletionStage<Map<String, IncidentDocument>> getIncidentDocuments(
+      final List<String> incidentIds) {
+    final var request = createIncidentDocumentsRequest(incidentIds);
+
+    return client
+        .search(request, IncidentEntity.class)
+        .thenApplyAsync(this::createIncidentDocuments, executor);
+  }
+
+  private SearchRequest createIncidentDocumentsRequest(final List<String> incidentIds) {
+    final var idQ = QueryBuilders.ids(i -> i.values(incidentIds));
+    final var partitionQ =
+        QueryBuilders.term(t -> t.field(IncidentTemplate.PARTITION_ID).value(partitionId));
+    return new SearchRequest.Builder()
+        .index(incidentAlias)
+        .query(q -> q.bool(b -> b.must(idQ, partitionQ)))
+        .allowNoIndices(true)
+        .ignoreUnavailable(true)
+        .build();
```

Comment:
:exclamation: I removed the sort by key from the original implementation. I tried to understand why it was there in the first place, and honestly it doesn't make much sense. I assume it's a historical leftover?

Sorting by key (since keys are monotonically generated) gives you _some_ indication of chronology - but I still don't see how it's relevant here in any way, or how it used to be before that.

Reply:
:thought_balloon:  Should we remove the sorting after we tested that this implementation is working as expected? This may help simplify testing it since we have done less changes. 
Removing the sorting afterwards is quite quick.

Reply:
Added the sort back just in case, you're right, easy to remove it later. I don't know if it has any sort of performance implications on the ES side (probably - but then again, unclear if it's important).